base/libgit2/config.jl

```diff
@@ -16,13 +23,27 @@ function GitConfig(path::AbstractString,
     return cfg
 end
 
+"""
+    GitConfig(repo::GitRepo)
+
+Get the stored configuration for the git repository `repo`. If `repo` does not
+have a specific configuration file set, the default git configuration will be
+used.
+"""
 function GitConfig(repo::GitRepo)
     cfg_ptr_ptr = Ref{Ptr{Void}}(C_NULL)
     @check ccall((:git_repository_config, :libgit2), Cint,
                   (Ptr{Ptr{Void}}, Ptr{Void}), cfg_ptr_ptr, repo.ptr)
     return GitConfig(repo, cfg_ptr_ptr[])
 end
 
+"""
+    GitConfig(level::Consts.GIT_CONFIG = Consts.CONFIG_LEVEL_DEFAULT)
+
+Get the default git configuration by loading the global, system, and XDG configuration
```

Comment:
It's some sort of weird file thing I don't understand. I think it's how git internally organizes its files?